pkg/operation/botanist/component/metricsserver/metrics_server.go

```diff
@@ -382,37 +428,6 @@ func (m *metricsServer) computeResourcesData() (map[string][]byte, error) {
 		})
 	}
 
-	if m.vpaEnabled {
-		vpaUpdateMode := autoscalingv1beta2.UpdateModeAuto
-		vpa = &autoscalingv1beta2.VerticalPodAutoscaler{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "metrics-server",
-				Namespace: metav1.NamespaceSystem,
-			},
-			Spec: autoscalingv1beta2.VerticalPodAutoscalerSpec{
-				TargetRef: &autoscalingv1.CrossVersionObjectReference{
-					APIVersion: appsv1.SchemeGroupVersion.String(),
-					Kind:       "Deployment",
-					Name:       deployment.Name,
-				},
-				UpdatePolicy: &autoscalingv1beta2.PodUpdatePolicy{
-					UpdateMode: &vpaUpdateMode,
-				},
-				ResourcePolicy: &autoscalingv1beta2.PodResourcePolicy{
-					ContainerPolicies: []autoscalingv1beta2.ContainerResourcePolicy{
-						{
-							ContainerName: autoscalingv1beta2.DefaultContainerResourcePolicy,
-							MinAllowed: corev1.ResourceList{
-								corev1.ResourceCPU:    resource.MustParse("50m"),
-								corev1.ResourceMemory: resource.MustParse("150Mi"),
-							},
-						},
-					},
-				},
-			},
-		}
-	}
```

Comment:
You would need to pro-actively delete the VPA object (it will still exist in the system for existing shoot clusters).

Reply:
@rfranzke Wouldn't the old resources be [cleaned up](https://github.com/gardener/gardener-resource-manager/blob/4d38c292feea2ad3d653f824ea659fb81bded192/pkg/controller/managedresource/reconciler.go#L284) by GRM?

Reply:
Yes, true, please ignore above comment